ls8/cpu.py

```diff
@@ -62,4 +76,26 @@ def trace(self):
 
     def run(self):
         """Run the CPU."""
-        pass
+        # Perform REPL style execution
+        running = True
+        while running:
+            ()
+            # Start the CPU. start storing instructions in IR
+            self.ir = self.ram_read(self.pc)
+            operand_a = self.ram_read(self.pc+1)
+            operand_b = self.ram_read(self.pc+2)
+            # Exit the program when Instruction HLT is read
+            if self.ir == self.instruction["HLT"]:
+                running = False
+                self.pc += 1
+            # Execute other instructions
+            elif self.ir == self.instruction["LDI"]:
+                # Immediate set value of register to an intger
+                self.reg[operand_a] = 8
+                self.pc += 3
+            elif self.ir == self.instruction["PRN"]:
+                print(self.reg[operand_a])
+                self.pc += 2
+            else:
+                print(f"Eroor: Unknown Command {self.ir}")
+                sys.exit(1)
```

Comment:
How could we improve this if-else block to be more extensible?

Reply:
As I was reading the Step 10, using something call branch which will give better performance python wise